pkg/plugin/plugin.go

```diff
@@ -559,6 +559,15 @@ func CmdDel(args *skel.CmdArgs) error {
 			}
 			return err
 		})
+		// do the following as per cni spec (i.e. Plugins should generally complete a DEL action
+		// without error even if some resources are missing)
+		if _, ok := err.(ns.NSPathNotExistErr); ok || err == ip.ErrLinkNotFound {
```

Comment:
Could you please add a test for these two cases? Use the CNI to ADD, then manually break it and finally call DEL.

The logic of the cleanup code has grown substantially and we should make sure it is covered with functional tests, to avoid regressions.

Reply:
added tests covering these two cases. pls have a look.